lib/server/db/seeds/connection.js

```diff
@@ -0,0 +1,98 @@
+/**
+ * Utility for adding configurable Connections seeds to the
+ * database.
+ *
+ * @module ConnectionSeed
+ */
+var User = require("../model/User");
+var Connection = require("../model/Connection");
+
+/**
+ * @typedef {Object} ConnectionSeedConfig
+ * @property {String} ConnectionSeedConfig.user - The from-User
+ * @property {String[]} ConnectionSeedConfig.friends - The to-Users
+ */
+
+/**
+ * Create connections between a user and a variable number of friends and
+ * returns a then-able Promise, passing the connections array to the next "then".
+ *
+ * @private
+ * @function createConnections
+ * @param {ConnectionSeedConfig} config - The name of the from-User
+ * @returns {Promise}
+ */
+function createConnections(config) {
+    return User
+        .findOne({
+            username: config.user
+        })
+        .then(function (user) {
```

Comment:
Promises are a way of flattening the callback-hell we were talking about earlier: [Promise Docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise).

Basically, they take those nested callbacks and flatten them by registering the "next" thing via calls to `then` and `catch`. You'll notice that you can pipe data from one call to `then` to the next. This allows us to run asynchronous calls synchronously, since each registered `then` function only runs after the last one has completed.